src/pages/Admin/Details/Form/DetailsForm.tsx

```diff
@@ -334,8 +278,77 @@ const DetailsForm: React.FC<DetailsFormProps> = ({ venue }) => {
     [dirty, isSaving, isSubmitting]
   );
 
+  const { ownedVenues } = useOwnedVenues({});
+
+  const venueSpaces = ownedVenues.map(({ name, template }) => ({
+    name,
+    template,
+  }));
+
+  const [userStatuses, setUserStatuses] = useState<UserStatus[]>(
+    values.userStatuses ?? []
+  );
+
+  const addUserStatus = useCallback(
+    () =>
+      setUserStatuses([
+        ...userStatuses,
+        { status: "", color: DEFAULT_USER_STATUS.color },
+      ]),
+    [userStatuses, setUserStatuses]
+  );
+
+  const deleteStatus = useCallback(
+    (index: number) => {
+      const statuses = [...userStatuses];
+      statuses.splice(index, 1);
+      setUserStatuses(statuses);
+    },
+    [userStatuses, setUserStatuses]
+  );
+
+  const changeInput = useCallback(
+    (event: React.FormEvent<HTMLInputElement>, index: number) => {
+      const statuses = [...userStatuses];
+
+      statuses[index] = {
+        color: statuses[index].color,
+        status: event.currentTarget.value,
+      };
+      setUserStatuses(statuses);
+    },
+    [userStatuses, setUserStatuses]
+  );
+
+  const pickColor = useCallback(
+    (color: string, index: number) => {
+      const statuses = [...userStatuses];
+      statuses[index] = { color, status: statuses[index].status };
+      setUserStatuses(statuses);
+    },
+    [userStatuses, setUserStatuses]
+  );
+
+  const renderVenueUserStatuses = useMemo(
+    () =>
+      userStatuses.map((userStatus, index) => (
+        <UserStatusPanel
+          key={`${userStatus}-${index}`}
+          userStatus={userStatus}
+          onPickColor={(color) => pickColor(color, index)}
+          onChangeInput={(value) => changeInput(value, index)}
+          onDelete={() => deleteStatus(index)}
+        />
+      )),
+    [changeInput, deleteStatus, pickColor, userStatuses]
+  );
+
+  const jukeboxToggleClasses = classNames({
+    "mod--hidden": venue?.template !== VenueTemplate.jazzbar,
+  });
+
   return (
-    <Form onSubmit={handleSubmit(setVenue)} className="DetailsForm">
+    <Form className="DetailsForm" onSubmit={handleSubmit(setVenue)}>
```

Comment:
🚀 Good job on the input fields, there is a bit of need for a space between the inputs and the form borders however. This is how it works in `SpaceEditFormNG`

```scss

.SpaceEditFormNG {
  .AdminSection {
    padding: $spacing--sm $spacing--xl;
  }
}

Reply:
40ab2db